### ISSUE TITLE
[Reproduction] Stage-1 and Stage-2 outputs are blurry/tiled — possible mismatch between LTX 0.9.8-distilled config and 0.9.7-dev weights

### ISSUE THREAD

**Issue by huohuohuohuohuohuohuohuo** [fyting/ArtifactWorld#3]
Hi authors,

Thank you for releasing ArtifactWorld, including the two-stage inference code, benchmark, LoRA weights, and auxiliary latents.

I followed the official inference pipeline using the released, non-quantized models. Both stages completed successfully without CUDA OOM or runtime errors. However, the Stage-1 heatmap output is nearly uniform and blurry, while the final Stage-2 restored video contains strong repetitive/tiled textures and does not provide meaningful restoration.

After checking the complete pipeline, I noticed a possible inconsistency between the declared model version, the model files actually loaded, and the inference parameters. I would appreciate your help confirming the intended configuration.

## Repository version

- Repository: `fyting/ArtifactWorld`
- Commit: `c80b39678ddc5dbd77a80ec534082512d506cfdb`
- Relevant configs and pipeline files were unchanged from this commit.
- The only local code change was an optional CPU-offload branch guarded by `ARTIFACTWORLD_MODEL_CPU_OFFLOAD=1`.
- CPU offload was disabled during this run, so this branch was not executed.

## Environment

- GPU: NVIDIA RTX A6000, 48 GB
- CUDA runtime: 12.4
- Python: 3.10
- PyTorch: `2.6.0+cu124`
- Diffusers: `0.33.1`
- Transformers: `4.52.2`
- Accelerate: `1.2.1`
- PEFT: `0.14.0`
- Precision: BF16
- Quantization: disabled
- Text encoder 8-bit loading: disabled
- CPU/model offload: disabled
- No CUDA OOM occurred

The environment versions match the released `requirements.txt`.

## Official files used

I downloaded and verified the released files without using quantized alternatives.

### Benchmark archives

- `artifact.tar`
  - SHA-256: `30bca150762c39178262b38552b716cccbd8b7f9ea2b379f93fed5bb80e294f1`
- `gt.tar`
  - SHA-256: `ebf0740037f8d89cdd44be2db7fa8badb6cce32bf8a93093863c0eddc11b590c`

### ArtifactWorld weights

- `weights.tar`
  - Size: `2047044416` bytes
  - SHA-256: `32399f86b714748032482e9438dc031ab5e0301e9e633ac4d4e1a4bb83943451`

The archive checksum and size match the official Hugging Face repository.

The following released files were used:

- `stage1_pred_headmap_lora.safetensors`
- `stage2_noisemap_blend_lora.safetensors`
- `auxiliary_latents/z_full.pt`
- `auxiliary_latents/z_null.pt`

Both LoRA files contain 960 adapter tensors. The auxiliary latent shape is `[6630, 128]`, which matches the latent grid for a 97-frame, 960×544 video.

The LTX tokenizer, text encoder, VAE, scheduler, and transformer files were also verified against their official repository metadata.

## Benchmark sample

The tested benchmark pair is:

```text
009ie95i0twn43nasep3.mp4
```

The artifact and GT files were extracted directly from the official archives and verified against the corresponding tar members.

This is a real pair from ArtifactWorld-Benchmark, although it is not one of the selected examples displayed on the project page.

## Preprocessing

I used the released preprocessing script without modification:

```bash
python tools/process_videos.py \
  --gt-dir /path/to/gt \
  --artifact-dir /path/to/artifact \
  --output-dir /path/to/processed
```

The generated input has:

- Resolution: `960 × 544`
- Frames: `97`
- Frames 0: GT first frame
- Frames 1–87: artifact middle frames
- Frame 88: GT last frame
- Frames 89–96: repeated GT last frame

I verified the frame mapping independently. Corresponding processed/source frames have approximately 40–43 dB PSNR, so the differences are only from the script’s CRF18 H.264 re-encoding.

## Commands

Stage 1:

```bash
CUDA_VISIBLE_DEVICES=0 ./scripts/run_stage1_infer.sh \
  --input-folder /path/to/processed \
  --output-folder /path/to/stage1_output
```

Stage 2:

```bash
CUDA_VISIBLE_DEVICES=0 ./scripts/run_stage2_infer.sh \
  --input-folder /path/to/processed \
  --noisemap-videos-folder /path/to/stage1_output/samples \
  --output-folder /path/to/stage2_output
```

Stage 2 successfully matched the reference and noisemap by filename:

```text
Found 1 reference video file(s) to process
Matched 1 noisemap video(s) with reference videos
```

The log also confirmed that both auxiliary latents and all eight three-way heatmap blend weights were enabled.

## Effective inference configuration

The released YAML config declares:

```yaml
model:
  model_source: LTXV_13B_098_DISTILL
```

However, the same config explicitly sets:

```yaml
scheduler_source: __LTX_097_DEV_ROOT__
transformer_source: __LTX_097_DEV_ROOT__
```

After placeholder expansion, the actual values used were:

```text
model.model_source       = LTXV_13B_098_DISTILL
model.scheduler_source   = /models/LTX-Video-0.9.7-dev
model.transformer_source = /models/LTX-Video-0.9.7-dev
validation.inference_steps = 8
validation.guidance_scale  = 1.0
```

In `load_ltxv_components()`, the explicit `transformer_source` and `scheduler_source` take precedence over `model_source`. Therefore, the transformer actually loaded during this run was the non-distilled `LTX-Video-0.9.7-dev`, despite `model_source` being labeled `LTXV_13B_098_DISTILL`.

The logs show six transformer shards from `LTX-Video-0.9.7-dev` being loaded.

There was also this scheduler warning:

```text
The config attributes {'stochastic_sampling': False} were passed to
FlowMatchEulerDiscreteScheduler, but are not expected and will be ignored.
```

### Stage 1

The generated artifact heatmap/noisemap is mostly a low-frequency brown/purple image with very little spatial correspondence to the severe artifact regions.

It does not look like an effective spatial artifact-intensity map for this sample.

### Stage 2

The restored video:

- is heavily blurred;
- contains repetitive/tiled grid-like textures;
- loses most scene structure;
- does not preserve the clean boundary content;
- is visually worse than the artifact input.

This is not only a video-player or transfer problem. I extracted and compared decoded frames directly.

### Quantitative comparison

Over the first 89 corresponding frames:

| Comparison | PSNR | SSIM |
|---|---:|---:|
| Artifact vs. GT | 13.623 | 0.458 |
| Restored vs. GT | 15.800 | 0.461 |

Although PSNR increases slightly because the output is strongly smoothed, SSIM remains essentially unchanged, and the restored video loses important geometry and high-frequency structure.

I also performed an isolated VAE encode/decode test using the same official VAE and the actual processed input:

```text
Input shape:   [1, 3, 97, 544, 960]
Latent shape:  [1, 128, 13, 17, 30]
Output shape:  [1, 3, 97, 544, 960]
VAE round-trip PSNR: 23.82 dB
```

The VAE round trip preserves recognizable content, so the catastrophic blur/tiled pattern appears to be introduced during diffusion inference rather than during video reading, VAE loading, or output encoding.

## Possible configuration mismatch

The LTX-Video documentation recommends approximately 8 steps and no CFG for distilled models. In contrast, the non-distilled dev model normally uses substantially more inference steps and guidance greater than 1.

This raises the following possibility:

```text
Declared model: 0.9.8 distilled
Actual transformer/scheduler: 0.9.7 dev
Sampling: 8 steps, guidance 1.0
```

Could the released LoRA adapters have been trained for the 0.9.8 distilled backbone, while the release config currently loads the 0.9.7-dev transformer?

Alternatively, if the intended backbone is 0.9.7-dev, are 8 inference steps and guidance 1.0 still the intended settings after LoRA fine-tuning?

## Questions

Could you please confirm the following?

1. Which exact base transformer was used to train LoRA1 and LoRA2?

   - `LTX-Video-0.9.7-dev`
   - `LTX-Video-0.9.7-distilled`
   - `LTX-Video-0.9.8-13B-distilled`
   - another revision/checkpoint

2. What are the exact Hugging Face repository and revision/commit hashes expected for:

   - transformer
   - scheduler
   - VAE
   - tokenizer
   - text encoder

3. Is the following combination intentional?

   ```text
   LTX-Video-0.9.7-dev transformer
   8 inference steps
   guidance scale 1.0
   ```

4. What should a valid Stage-1 heatmap look like?

   Should the decoded Stage-1 video be directly interpretable as an artifact-intensity map, or is its visible RGB appearance not meaningful?

5. Could you provide one canonical benchmark filename and its expected Stage-1/Stage-2 outputs?

   Even a single example filename, preview image, or output checksum would make it possible to distinguish a local environment problem from a released configuration/model mismatch.

6. Is `009ie95i0twn43nasep3.mp4` expected to work with the released checkpoint?

I can provide the complete Stage-1/Stage-2 logs, processed input, heatmap output, restored output, and frame comparisons if useful.

<img width="1440" height="544" alt="Image" src="https://github.com/user-attachments/assets/e582f41e-3604-408b-8aa8-7468fded0255" />
the upper are artifact、GT、processed input；the bottom are Stage‑1 heatmap、Stage‑2 restored

[artifactworld-finish.log](https://github.com/user-attachments/files/30340803/artifactworld-finish.log)
[artifactworld-stage1-official.log](https://github.com/user-attachments/files/30340807/artifactworld-stage1-official.log)
[artifactworld-stage2-official.log](https://github.com/user-attachments/files/30340806/artifactworld-stage2-official.log)
[stage1_infer.yaml](https://github.com/user-attachments/files/30340804/stage1_infer.yaml)
[stage2_infer.yaml](https://github.com/user-attachments/files/30340805/stage2_infer.yaml)

Thank you very much for your help and for releasing the project.
